### ISSUE TITLE
Watchman integration giving wrong results when parent directory is in watch-list

### ISSUE THREAD

**Issue by lencioni** [Galooshi/import-js#337]
I haven't had time to fully track this down yet, but I've been noticing some odd behavior when using import-js with watchman integration.

Specifically, I keep my repos in my home directory. At a fresh start, things are working just fine.

At some later point, I notice that the results that import-js is finding aren't correct. Instead of something like "../path/to/thing" the module name is more like "../../project/path/to/thing". This is additionally weird because that path isn't even correctly relative to the current file--in this case it should be "../../../project/path/to/thing".

And sometimes, after this point, when I run fix imports, vim ends up locking up and taking 100% CPU. The only way to recover is to `kill -9` the process. Import word does not cause vim to have this CPU usage and freeze, but it does have the strange paths. This might be a separate issue.

I've noticed that my home directory appears in the watch-list when this happens. I'm not entirely sure why, or why it might have an effect. Also, the 100% CPU usage may be unrelated (I also happen to be running the remove-strip-from-path branch locally, which may be part of this). 

```
~/project ❯❯❯ watchman watch-list
{
    "version": "4.6.0",
    "roots": [
        "/Users/joe_lencioni",
        "/Users/joe_lencioni/project",
    ]
}
```

If I remove my home directory from the watch list, the paths start working correctly again. However, I still end up with 100% CPU sometimes (which makes me further think that is a separate issue).

**Comment by lencioni** [comment]
I wonder if any of this has anything to do with me breaking `importjsd --parent-pid #`

**Comment by trotzig** [comment]
I doubt it has anything to do with that. But there might have been a bug at some point causing the root folder to be hooked up. 

Here are the folders I watch currently:

```
{
    "version": "4.5.0",
    "roots": [
        "/Users/henrictrotzig/react-simple-pie-chart",
        "/Users/henrictrotzig/vim-import-js",
        "/Users/henrictrotzig/brigade",
        "/Users/henrictrotzig/causes",
        "/Users/henrictrotzig/import-js"
    ]
}
```

Have you tried restarting the watchman service?

**Comment by trotzig** [comment]
We resolved this offline. The issue was (if I recall correctly) that watchman had reached its match file limit, because of how the home folder was watched. 

Since this was reported, import-js has code in place to prevent enabling watchman for certain folders, which should prevent the home folder from getting watched.

**Comment by lencioni** [comment]
Specifically, I think this was fixed by 0f9ed418b3d27e2eefac3936d45733c04a37f2fa.

**Comment by trotzig** [comment]
👍 thanks for digging it up. I was lazy...
